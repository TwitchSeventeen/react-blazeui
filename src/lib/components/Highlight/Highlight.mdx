---
name: Highlight
---

import { Playground, PropsTable } from 'docz'
import Highlight from './index';

# Highlight

This adds a highlight to text. It can be used inline (ie it's not a block level element).

## Properties
<PropsTable of={Highlight} />

## Basic usage
<Playground>
    <Highlight>
        Use the Highlight component to highlight text
    </Highlight>
</Playground>