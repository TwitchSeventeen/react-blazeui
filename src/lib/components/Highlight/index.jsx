import React from 'react';
import PropTypes from 'prop-types';

function Highlight (props) {
  const { children } = props;
  return (
      <span className="u-text--highlight">
        { children }
      </span>
  );
}

Highlight.propTypes = {
  /**
   * This element can have children
   */
  children: PropTypes.node,
};

Highlight.defaultProps = {
  children: null,
};

export default Highlight;