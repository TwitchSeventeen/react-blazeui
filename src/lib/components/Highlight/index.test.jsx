import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Highlight from './index';

describe('Highlight', () => {
  it('should be defined', () => {
    expect(Highlight).toBeDefined();
  });

  it('renders correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
        <Highlight>
          lorem ipsum dolor
        </Highlight>
    );

    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });
});
