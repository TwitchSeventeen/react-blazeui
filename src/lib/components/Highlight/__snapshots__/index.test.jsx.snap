// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Highlight renders correctly 1`] = `
<span
  className="u-text--highlight"
>
  lorem ipsum dolor
</span>
`;
